build/driver.go

```diff
@@ -19,6 +19,8 @@ import (
 )
 
 type resolvedNode struct {
+	SolveOpt *client.SolveOpt
```

Comment:
I think @jedevc did not like the `driverPair` name, but this is the use case for why it was added. That we pass along a struct that contains both the driver (in here it is called resolver+platform, so driver for a specific node) and the request.

I think even if we use embed structs, the separation of structs would be clearer this way.

Reply:
After discussing with @tonistiigi, the idea is to have a clear separation of concerns between drivers resolution and solve opts attached to the driver in build. Pushed an extra commit to move `*client.SolveOpt` to a local struct type in build.